File: frontend/src/objects/merchant/apiTypes/UpdateProductRequest.ts
import type { ListingStatus } from '@/objects/shared/ids'
import type { ProductBundleGroup, ProductInventoryMode } from '@/objects/merchant/Product'

export interface UpdateProductRequest {
  name: string
  description: string
  imageUrl: string
  categoryName: string
  price: number
  remainingStock: number
  listingStatus: ListingStatus
  inventoryMode?: ProductInventoryMode
  maxPerOrder?: number | null
  bundleGroups?: ProductBundleGroup[]
}

File: backend/src/merchant/objects/apiTypes/UpdateProductRequest.scala
package delivery.merchant.objects.apiTypes

import delivery.merchant.objects.ProductBundleGroup
import delivery.shared.objects.ListingStatus

final case class UpdateProductRequest(
    name: String,
    description: String,
    imageUrl: String,
    categoryName: String,
    price: Double,
    remainingStock: Int,
    listingStatus: ListingStatus,
    inventoryMode: Option[String] = None,
    maxPerOrder: Option[Int] = None,
    bundleGroups: Option[List[ProductBundleGroup]] = None
)
